meilisearch-http/src/index/mod.rs

```diff
@@ -51,11 +52,22 @@ impl Index {
             .map(|c| c.to_string())
             .collect();
 
+        let stop_words = self
+            .stop_words(&txn)?
+            .map(|stop_words| {
+                std::iter::repeat(()).scan(stop_words.stream(), |stream, _| {
+                    stream.next().map(|s| String::from_utf8(s.to_vec()))
+                })
+                .collect::<Result<BTreeSet<String>, _>>()
+            })
+            .transpose()?;
+
```

Comment:
I think that this makes the code a bit obscure. I would go with a simple solution, at the expense of an extra allocation.

Also we  want to return an empty array when there are no stop words, since this is the current meilisearch behaviour (I assume milli returns None when there are no stop-words?).

```suggestion
        let stop_words = self
            .stop_words(&txn)?
            .map(|stop_words| -> anyhow::Result<BTreeSet<_> {
                Ok(stop_words
                .stream()
                .into_strs()?
                .into_iter()
                .collect())
            })
            .transpose()?
            .unwrap_or_else(BTreeSet::new);

```